src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -776,22 +775,41 @@ const parseKey = debounce(500, (key, setKeyObject, setKeyInvalid) => {
                 setKeyInvalid(true);
                 console.warn("Could not validate the public key");
             });
-});
+};
+
+const handleSSHKeysValue = (value, idx, onChange, additem) => {
+    let index = idx;
+    for (const part of value.split("\n")) {
+        if (part.trim() === "") {
+            continue;
+        }
+        if (index !== idx) {
+            additem();
+        }
+
+        onChange(index, "value", part);
+        index++;
+    }
+};
 
 const SshKeysRow = ({
-    id, item, onChange, idx, removeitem,
+    id, item, onChange, idx, removeitem, additem
 }) => {
     const [keyObject, setKeyObject] = useState();
     const [keyInvalid, setKeyInvalid] = useState(false);
+    const keyValue = item.value || "";
 
-    const onChangeHelper = (value) => {
-        // Some users might want to input multiple keys into the same field
-        // While handling that in the future might be a nice user experience, now we only parse one key out of input
-        value = value.split(/\r?\n/)[0];
-
-        onChange(idx, "value", value);
-        parseKey(value, setKeyObject, setKeyInvalid);
-    };
+    useEffect(() => {
+        let timeoutId = null;
+        if (keyValue !== "" && !keyObject) {
+            timeoutId = setTimeout(() => parseKey(keyValue, setKeyObject, setKeyInvalid), 500);
```

Comment:
I think the [original debounce comment](https://github.com/cockpit-project/cockpit-machines/pull/1309/files#diff-c3b5a228e7299618756cc30089b5e5f0b1e3662e48459df036e31676a874bec2L755) still applies here -- this useEffect() re-runs on *every* input, so it creates multiple parallel timers and calls parseKey() for each one as well. Some debouncing still needs to happen.

Reply:
The issue with a global debounce is that it is used for multiple components, so if you add two keys quickly only one will get parsed as the other call is debounced. So debouncing needs to happen per Component and we can't keep the debounce function in the component as every re-render it would be re-instantiated. So I believe the current approach with setTimeout debounces correctly, when a new as a new value would cause a re-render so clearTimeout() to be called?

If I'm wrong I'm happy to fix it if there is a better way.

Reply:
We can not make onChange call setKeyValue simply because onChange will add a new row if multiple keys are added using `additem()`. That is the hard part, dynamically creating a new row component.

Reply:
That sounds weird -- in JSX it's just a single TextArea. If this somehow multiplies the TextArea, this needs fixing -- either we shouldn't multiply it, or have a full-blown dynamic set of input lines where each one just has a *single* key, and :heavy_plus_sign: :wastebasket: buttons